lib/tests/streamlit/type_util_test.py

```diff
@@ -661,16 +661,110 @@ def test_convert_df_with_missing_values(self):
+            B = enum.auto()
+            C = enum.auto()
+
+        EnumAOrig = EnumA
+
+        class EnumA(enum.Enum):
+            A = enum.auto()
+            B = enum.auto()
+            C = enum.auto()
+
+        EnumAEqual = EnumA
+
+        class EnumA(enum.Enum):
+            A = enum.auto()
+            B = enum.auto()
+            D = enum.auto()
+
+        EnumADiffMembers = EnumA
+
+        class EnumA(enum.Enum):
+            A = "1"
+            B = "2"
+            C = "3"
+
+        EnumADiffValues = EnumA
+
+        class EnumA(enum.Enum):
+            A = enum.auto()
+            B = enum.auto()
+            C = enum.auto()
+            D = enum.auto()
+
+        EnumAExtraMembers = EnumA
+
+        class EnumA(enum.Enum):
+            A = enum.auto()
+            B = enum.auto()
+            C = enum.auto()
+
+        EnumADiffQualname = EnumA
+        EnumADiffQualname.__qualname__ = "foobar.EnumA"
+
+        class EnumB(enum.Enum):
+            A = enum.auto()
+            B = enum.auto()
+            C = enum.auto()
+
+        assert all(
+            EnumAOrig.A not in enum
+            for enum in (
+                EnumAEqual,
+                EnumADiffMembers,
+                EnumADiffValues,
+                EnumADiffQualname,
+            )
+        )
+        assert EnumAOrig.A.value == EnumAEqual.A.value
+
+        # Things that are coercable
+        assert type_util.coerce_enum(EnumAOrig.A, EnumAEqual) is EnumAEqual.A
+        assert type_util.coerce_enum(EnumAOrig.A, EnumADiffValues) is EnumADiffValues.A
+        # Things that are not coercable
+        assert type_util.coerce_enum(EnumAOrig.A, EnumADiffMembers) is EnumAOrig.A
+        assert type_util.coerce_enum(EnumAOrig.A, EnumAExtraMembers) is EnumAOrig.A
+        assert type_util.coerce_enum(EnumAOrig.A, EnumB) is EnumAOrig.A
+        assert type_util.coerce_enum(EnumAOrig.A, EnumADiffQualname) is EnumAOrig.A
+        # Things that do not require coercion
+        assert type_util.coerce_enum(EnumAOrig.A, EnumAOrig) is EnumAOrig.A
+        # Things which cause errors
+        with pytest.raises(ValueError, match="Expected an EnumMeta"):
+            type_util.coerce_enum(EnumAOrig.A, EnumAEqual.A)
+        with pytest.raises(ValueError, match="Expected an Enum"):
+            type_util.coerce_enum(EnumAOrig, EnumAEqual)
+
+        # test with different config option
+        with patch_config_options({"runner.enumCoercion": "off"}):
+            assert type_util.coerce_enum(EnumAOrig.A, EnumAEqual) is EnumAOrig.A
+        with patch_config_options({"runner.enumCoercion": "nameAndValue"}):
+            assert type_util.coerce_enum(EnumAOrig.A, EnumAEqual) is EnumAEqual.A
+            assert type_util.coerce_enum(EnumAOrig.A, EnumADiffValues) is EnumAOrig.A
+        with pytest.raises(errors.StreamlitAPIException), patch_config_options(
+            {"runner.enumCoercion": "badValue"}
+        ):
+            type_util.coerce_enum(EnumAOrig.A, EnumAEqual)
```

Comment:
Similar comment here as in a previous file -- I think it'd be more readable to split this and the setup above into its own new test class and use test names to describe what's going on rather than comments

Reply:
So:
* Leave in type_util_test.py
* Make a `class TestEnumCoersion`
* Make each test "section" here a separate `test_method()`

yes @vdonato?